nin/frontend/app/scripts/controllers/main.js

```diff
@@ -255,23 +265,34 @@
                 var layer = $scope.layers[i];
                 if (layerShaderDependencies[layer.type]) {
                   if (layerShaderDependencies[layer.type].indexOf(event.shadername) !== -1) {
-                    demo.lm.refresh(layer.type);
+                    demo.nm.refresh(layer.type);
                   }
                 }
               }
 
-              demo.lm.update(demo.looper.currentFrame);
+              demo.nm.update(demo.looper.currentFrame);
               Loader.start(function() {}, function() {});
               break;
 
-            case 'layer':
-              var indirectEval = eval;
+            case 'node':
               $scope.fileCache[event.path] = event.content;
+              var indirectEval = eval;
               indirectEval(event.content);
-              layerShaderDependencies[event.layername] = event.shaderDependencies;
 
-              demo.lm.refresh(event.layername);
-              demo.lm.update(demo.looper.currentFrame);
+              var splitted = event.path.split('/');
+              var filename = splitted[splitted.length - 1];
+              var typename = filename.slice(0, -3);
```

Comment:
Exactly 2, actually, since the third character is `.`. Luckily, since this only handles .js-files, it should be ok.